src/Octoshift/FileSystemProvider.cs

```diff
@@ -26,4 +26,12 @@ public virtual async ValueTask WriteAsync(FileStream fileStream, ReadOnlyMemory<
 
         await fileStream.WriteAsync(buffer, cancellationToken);
     }
+
+    public virtual void Delete(string path)
```

Comment:
What do you think of renaming this function so it's obvious what happens if the file isn't found?

```suggestion
    public virtual void DeleteIfExists(string path)
```

Reply:
addressed in 0bf2055c9a3573c78aa0dd5fee98ef35b8fa4e8c